irony-cdb.el

```diff
@@ -114,6 +128,57 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--choose-closest-path (file paths)
+  "Find the \"best\" path in PATHS matching FILE
+
+If any paths in PATHS is belongs to the same directory
+or a subdirectory of file, we disregard other candidates.
+
+For remaining candidates, \"nearest\" is measured as abs. difference
+in path depth.
+- We prefer deeper paths at level +N to those at level -N.
+- If multiple paths are equally good, we return the last one.
+
+Returns nil if paths isn't a list of at least one element.
+"
+  (when (listp paths)
+    (let ((paths (or
+                  ;; if we find a cdb in cwd or below, don't consider other candidates
+                  (cl-remove-if-not (lambda (x) (string-prefix-p (file-name-directory file) x)) paths)
+                  paths)))
+      (cl-loop for path in paths
+         with best-depth-delta = 999999 ; start at +inf
+         with best-path = nil ; we keep the best so far here
+         ;; all candidates have their depth compared to that of target file
+         with file-depth = (length (split-string file "/")) ;
```

Comment:
Nope, that's an emacs-style path-seperator:

https://www.gnu.org/software/emacs/manual/html_node/elisp/File-Names.html

     On MS-DOS and MS-Windows, these functions (like the function that actually operate 
     on files) accept MS-DOS or MS-Windows file-name syntax, where backslashes separate 
     the components, as well as Unix syntax; **but they always return Unix syntax**. This 
     enables Lisp programs to specify file names in Unix syntax and work properly on all 
     systems without change.

`file` here is the result of `(buffer-file-name)` and paths come from `locate-dominating-file`.